core/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTest.java

```diff
@@ -0,0 +1,274 @@
+                fields.put(randomAsciiOfLengthBetween(1, 10), 1.0f);
+            } else {
+                fields.put(randomAsciiOfLengthBetween(1, 10), 2.0f / randomIntBetween(1, 20));
+            }
+        }
+        SimpleQueryStringBuilder result = new SimpleQueryStringBuilder(text);
+        result.boost(boost).fields(fields).analyzer(analyzer).queryName(queryName).defaultOperator(operator);
+        result.minimumShouldMatch(minimumShouldMatch).locale(locale).lowercaseExpandedTerms(lowercaseExpandedTerms);
+        result.lenient(lenient).analyzeWildcard(analyzeWildcards).flags(flags);
+        return result;
+    }
+
+    // Check operator handling, and default field handling.
+    @Test
+    public void testDefaultOperatorHandling() {
+        SimpleQueryStringBuilder qb = new SimpleQueryStringBuilder("The quick brown fox.");
+        BooleanQuery boolQuery = (BooleanQuery) qb.toQuery(createContext());
+        assertThat(shouldClauses(boolQuery), is(4));
+
+        qb.defaultOperator(Operator.AND);
+        boolQuery = (BooleanQuery) qb.toQuery(createContext());
+        assertThat(shouldClauses(boolQuery), is(0));
+
+        qb.defaultOperator(Operator.OR);
+        boolQuery = (BooleanQuery) qb.toQuery(createContext());
+        assertThat(shouldClauses(boolQuery), is(4));
+    }
+
+    @Test
+    public void testValidation() {
+        SimpleQueryStringBuilder qb = createTestQueryBuilder();
+        assertNull(qb.validate());
+    }
+
+    public void testNullQueryTextGeneratesException() {
+        SimpleQueryStringBuilder builder = new SimpleQueryStringBuilder(null);
+        QueryValidationException exception = builder.validate();
+        assertThat(exception, notNullValue());
+    }
+
+    @Test
+    public void testHandlingDefaults() throws IOException {
+        SimpleQueryStringBuilder qb = createTestQueryBuilder();
+        qb.analyzer(null);
+        qb.analyzeWildcard(null);
+        qb.defaultOperator(null);
+        qb.field(null);
+        qb.flags((SimpleQueryStringFlag[]) null);
+        qb.lenient(null);
+        qb.locale(null);
+        qb.lowercaseExpandedTerms(null);
+        qb.minimumShouldMatch(null);
+        qb.queryName(null);
+        assertEquals(qb.toQuery(createContext()), createExpectedQuery(qb, createContext()));
+    }
+
+    /*
+     * This assumes that Lucene query parsing is being checked already, adding
+     * checks only for our parsing extensions.
+     * 
+     * Also this relies on {@link SimpleQueryStringTests} to test most of the
+     * actual functionality of query parsing.
+     */
+    @Override
+    protected void assertLuceneQuery(SimpleQueryStringBuilder queryBuilder, Query query, QueryParseContext context) {
+        if (queryBuilder.queryName() != null) {
+            Query namedQuery = context.copyNamedFilters().get(queryBuilder.queryName());
+            assertThat(namedQuery, equalTo(query));
+        }
+
+        if (query instanceof BooleanQuery) {
```

Comment:
I am on the fence about these types of, where our checks depend on the output of the test (what we are testing!). 
I would rather change the if to verify that we get a boolean query when we expect it etc. and go ahead with the corresponding checks. Makes sense?